compiler/optimizer/OMRLocalCSE.cpp

```diff
@@ -280,7 +280,7 @@ void OMR::LocalCSE::transformBlock(TR::TreeTop * entryTree, TR::TreeTop * exitTr
    _numNodes = 0;
    _numNullCheckNodes = 0;
    _numCopyPropagations = 0;
-   _arrayRefNodes.deleteAll();
+   _arrayRefNodes = new (stackMemoryRegion) TR_ScratchList<TR::Node>(trMemory());
```

Comment:
So the arrayRefNodes was originally allocating out of the stack region for the optimization (rather than the per block stack region wrapped around transformBlock. As a result, the deleteAll memory was leaked until the end of the optimization. This change allocates a new list, but it is allocated out of a fresh scratch region all the time meaning that the memory will not live beyond the transform block leading to a lower memory footprint overall.